prisma/schema.prisma

```diff
@@ -0,0 +1,126 @@
+// This is your Prisma schema file,
+// learn more about it in the docs: https://pris.ly/d/prisma-schema
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+// ================ MODELS ================
+model User {
+  uid String @id
+
+  updatedAt DateTime @updatedAt @map("updated_at")
+
+  applications Application[]
+
+  @@map("users")
+}
+
+model Company {
+  id         Int     @id @default(autoincrement())
+  name       String
+  companyUrl String  @map("company_url")
+  isVerified Boolean @default(false) @map("is_verified")
+
+  updatedAt DateTime @updatedAt @map("updated_at")
+
+  roles Role[]
+
+  @@map("companies")
+}
+
+enum RoleType {
+  WINTER_INTERNSHIP
+  SPRING_INTERNSHIP
+  SUMMER_INTERNSHIP
+  FALL_INTERNSHIP
+  FULL_TIME
+}
+
+model Role {
+  id         Int      @id @default(autoincrement())
+  company    Company  @relation(fields: [companyId], references: [id])
+  companyId  Int      @map("company_id")
+  title      String
+  type       RoleType
+  year       Int
+  isVerified Boolean  @default(false) @map("is_verified")
+
+  updatedAt DateTime @updatedAt @map("updated_at")
+
+  applications Application[]
+
+  @@map("roles")
+}
+
+model Application {
+  id     Int    @id @default(autoincrement())
+  user   User   @relation(fields: [userId], references: [uid])
+  userId String @map("user_id")
+  role   Role   @relation(fields: [roleId], references: [id])
```

Comment:
Should we change to ON DELETE CASCADE for this?

Reply:
I don't believe we should allow roles to be deleted if applications exist. I don't think we should cascade delete from Role (admin access) to affect apps (user data).

Reply:
Yup, added cascade on updates and deletes

Reply:
I personally think it's fine to have `ON DELETE CASCADE`. `ON DELETE CASCADE` just makes it more dangerous if we actually do delete roles, but then again this is a rare case and I think for now we can assume that the admin takes active steps to protect user data before running the delete command.